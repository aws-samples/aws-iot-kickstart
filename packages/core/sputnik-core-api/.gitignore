lib/
generated/
schema.graphql
!bin/**
