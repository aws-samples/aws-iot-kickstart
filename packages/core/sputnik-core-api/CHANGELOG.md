# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2020-08-09)


### Features

* implement batch device deploy api/infra ([4a2ae89](https://git-codecommit.us-west-2.amazonaws.com/v1/repos/Deathstar/commits/4a2ae89f7f3f734521cf738683fcaddd2454dd78))
* **sputnik-core-api:** copy graphql operation files from ui in core api ([2ac85c0](https://git-codecommit.us-west-2.amazonaws.com/v1/repos/Deathstar/commits/2ac85c0a1e045bf416a0c90d22afddc1f0778745))
* **sputnik-core-api:** initial core api implementation ([38132c7](https://git-codecommit.us-west-2.amazonaws.com/v1/repos/Deathstar/commits/38132c72e9d7d2e5ff7309f004580ec8e78c930d))
