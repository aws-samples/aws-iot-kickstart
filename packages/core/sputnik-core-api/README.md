# `@deathstar/sputnik-core-api`

> TODO: description

Sputnik core api schema, types, operation, etc

## Usage

```
import api from '@deathstar/sputnik-core-api';

// TODO: Add usage details
```
