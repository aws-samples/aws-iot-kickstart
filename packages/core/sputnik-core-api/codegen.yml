overwrite: true
schema:
  [
    './src/schema/appsync.graphql',
    './lib/schema.graphql',
  ]
documents: 'src/operation/**/*.graphql'
generates:
  generated/types.ts:
    plugins:
      - 'typescript'
  generated/apollo-angular.ts:
    preset: import-types
    presetConfig:
      typesPath: ./types
    plugins:
      - 'typescript-operations'
      - 'typescript-apollo-angular'
    config:
      serviceName: SputnikApiService
      sdkClass: true
  generated/graphql-request.ts:
    preset: import-types
    presetConfig:
      typesPath: ./types
    plugins:
      - 'typescript-operations'
      - 'typescript-graphql-request'
