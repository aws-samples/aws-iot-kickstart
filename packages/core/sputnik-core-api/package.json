{
  "name": "@deathstar/sputnik-core-api",
  "description": "Sputnik core api schema, types, operation, etc",
  "version": "0.0.0-beta.0",
  "license": "Apache-2.0",
  "publishConfig": {
    "access": "public",
    "tag": "beta"
  },
  "author": "Jeremy Jonas <jay.jonas@hotmail.com>",
  "contributors": [
    "Timothee Cruse <timothee.cruse@gmail.com>",
    "Jeremy Jonas <jay.jonas@hotmail.com>",
    "Greg Varga <contact@gergelyvarga.com>",
    "Mirash Gjolaj <mirash.gjolaj@gmail.com>"
  ],
  "directories": {
    "lib": "lib",
    "test": "test"
  },
  "files": [
    "lib",
    "schema.graphql"
  ],
  "main": "lib/index.js",
  "typings": "lib/types.d.ts",
  "scripts": {
    "clean": "rm -rf lib generated",
    "prebuild": "yarn clean && yarn generate",
    "build": "tsc --project tsconfig.json",
    "postbuild": "cat src/schema/appsync.graphql > lib/schema.dev.graphql && cat lib/schema.graphql >> lib/schema.dev.graphql",
    "test": "jest",
    "generate": "yarn generate:schema && yarn generate:code && yarn generate:src",
    "generate:src": "cp src/schema-file.ts generated/schema-file.ts",
    "generate:schema": "ts-node ./bin/generate-schema.js",
    "generate:code": "graphql-codegen --config codegen.yml",
    "prewatch": "yarn clean && yarn generate:schema",
    "watch:schema": "nodemon -e ts,graphql -w src/schema -x yarn generate:schema",
    "watch:code": "yarn generate:code --watch",
    "watch": "run-p -c watch:*"
  },
  "dependencies": {
    "apollo-angular": "^1.10.0",
    "apollo-cache": "^1.3.5",
    "apollo-client": "^2.6.10",
    "graphql": "^15.3.0",
    "graphql-request": "^2.0.0"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^1.17.6",
    "@graphql-codegen/import-types-preset": "^1.17.6",
    "@graphql-codegen/typescript": "^1.17.6",
    "@graphql-codegen/typescript-apollo-angular": "^1.17.0",
    "@graphql-codegen/typescript-graphql-request": "^1.17.6",
    "@graphql-codegen/typescript-operations": "^1.17.6",
    "@graphql-codegen/typescript-react-apollo": "^1.17.6",
    "appsync-schema-converter": "^0.1.3",
    "graphql-tools": "^6.0.12",
    "graphql.macro": "^1.4.2",
    "nodemon": "^2.0.4",
    "ts-node": "^8.10.2"
  }
}
