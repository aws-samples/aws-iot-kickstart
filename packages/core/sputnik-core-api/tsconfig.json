{
  "extends": "../../../configs/workspace/tsconfig.json",
  "compilerOptions": {
    "noEmit": false,
    "emitDeclarationOnly": false,
    "skipLibCheck": true,
    "experimentalDecorators": true,
    "declaration": true,
    "declarationDir": "./lib",
    "rootDir": "./generated",
    "outDir": "./lib"
  },
  "compileOnSave": true,
  "exclude": ["node_modules", "dist"],
  "include": ["generated"],
}
