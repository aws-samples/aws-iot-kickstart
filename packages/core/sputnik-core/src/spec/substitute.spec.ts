import { SpecDefinition } from '@deathstar/sputnik-core-api'
import { substituteSpec } from './substitute'

// TODO: move this to global
type JestExpect = <R>(actual: R) => jest.Matchers<R> & jasmine.Matchers<R>;
declare const expect: JestExpect

describe('substitue', () => {
	test('should substitute tokens', () => {
		const substitutions = {
			CERTIFICATE_ARN: '!!cert-arn!!',
			THING_ARN: '!!thing-arn!!',
			FUNCTION_ARN: '!!function-arn!!',
			ACCOUNT: '!!acount!!',
		}

		const spec: SpecDefinition = {
			CoreDefinitionVersion: {
				Cores: [
					{
						CertificateArn: '[CERTIFICATE_ARN]',
						Id: 'Id',
						ThingArn: '[THING_ARN]',
					},
				],
			},
			DeviceDefinitionVersion: {
				Devices: [
					{
						CertificateArn: '[CERTIFICATE_ARN]',
						Id: 'Id',
						ThingArn: '[THING_ARN]',
					},
				],
			},
			FunctionDefinitionVersion: {
				Functions: [
					{
						FunctionArn: '[FUNCTION_ARN]',
						FunctionConfiguration: {},
						Id: '[ACCOUNT]-Id',
					},
				],
			},
		}

		expect(substituteSpec(spec, substitutions)).toMatchSnapshot('substitutions')
	})
})
