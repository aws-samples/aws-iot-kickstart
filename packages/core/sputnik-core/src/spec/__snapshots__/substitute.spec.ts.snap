// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`substitue should substitute tokens: substitutions 1`] = `
Object {
  "CoreDefinitionVersion": Object {
    "Cores": Array [
      Object {
        "CertificateArn": "!!cert-arn!!",
        "Id": "Id",
        "ThingArn": "!!thing-arn!!",
      },
    ],
  },
  "DeviceDefinitionVersion": Object {
    "Devices": Array [
      Object {
        "CertificateArn": "!!cert-arn!!",
        "Id": "Id",
        "ThingArn": "!!thing-arn!!",
      },
    ],
  },
  "FunctionDefinitionVersion": Object {
    "Functions": Array [
      Object {
        "FunctionArn": "!!function-arn!!",
        "FunctionConfiguration": Object {},
        "Id": "!!acount!!-Id",
      },
    ],
  },
}
`;
